### PR TITLE
Change internal representation of ecma-values to real ecma-values.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c [hunk 1]
```diff
@@ -356,13 +356,9 @@ ecma_builtin_number_prototype_object_value_of (ecma_value_t this_arg) /**< this
...
       ecma_property_t *prim_value_prop_p = ecma_get_internal_property (obj_p,
-                                                                       ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE);
+                                                                       ECMA_INTERNAL_PROPERTY_ECMA_VALUE);
 
-      ecma_number_t *prim_value_num_p;
-      prim_value_num_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_number_t,
-                                                          ecma_get_internal_property_value (prim_value_prop_p));
-
-      return ecma_make_number_value (*prim_value_num_p);
+      return ecma_copy_value (ecma_get_internal_property_value (prim_value_prop_p));
```
Comment (akiss77) [inline jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c]: No assert here that `ecma_get_internal_property_value (prim_value_prop_p)` is indeed a number?

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 2]
```diff
@@ -164,7 +161,7 @@ ecma_builtin_init_object (ecma_builtin_id_t obj_builtin_id, /**< built-in ID */
 
       ecma_property_t *prim_value_prop_p;
       prim_value_prop_p = ecma_create_internal_property (object_obj_p,
-                                                         ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE);
+                                                         ECMA_INTERNAL_PROPERTY_DATE_FLOAT);
```
Comment (akiss77) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c]: Cannot we have 
```c
ecma_set_internal_property_value (prim_value_prop_p, prim_prop_num_value_p);
```
here as well?

Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c]: No because prim_prop_num_value_p is a pointer not an ecma_value.

**Comment by LaszloLango** [comment]
What is the benefit or aim of this PR? Could you provide some description?

**Comment by zherczeg** [comment]
Patch updated with better description, and the assert is added as well.

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/ecma/base/ecma-gc.c [hunk 3]
```diff
@@ -242,9 +242,8 @@ ecma_gc_mark_property (ecma_property_t *property_p) /**< property */
 
       switch (ECMA_PROPERTY_GET_INTERNAL_PROPERTY_TYPE (property_p))
       {
-        case ECMA_INTERNAL_PROPERTY_PRIMITIVE_STRING_VALUE: /* compressed pointer to a ecma_string_t */
-        case ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE: /* compressed pointer to a ecma_number_t */
-        case ECMA_INTERNAL_PROPERTY_PRIMITIVE_BOOLEAN_VALUE: /* a simple boolean value */
+        case ECMA_INTERNAL_PROPERTY_ECMA_VALUE: /* an ecma_value_t except object */
+        case ECMA_INTERNAL_PROPERTY_DATE_FLOAT: /* compressed pointer to a ecma_number_t */
```
Comment (akiss77) [inline jerry-core/ecma/base/ecma-gc.c]: This is not necessarily a compressed pointer: on 32 bit archs this stores a native ptr.

File: jerry-core/ecma/base/ecma-globals.h [hunk 4]
```diff
@@ -186,9 +186,8 @@ typedef enum
   ECMA_INTERNAL_PROPERTY_NATIVE_CODE, /**< native handler location descriptor */
   ECMA_INTERNAL_PROPERTY_NATIVE_HANDLE, /**< native handle associated with an object */
   ECMA_INTERNAL_PROPERTY_FREE_CALLBACK, /**< object's native free callback */
-  ECMA_INTERNAL_PROPERTY_PRIMITIVE_STRING_VALUE, /**< [[Primitive value]] for String objects */
-  ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE, /**< [[Primitive value]] for Number objects */
-  ECMA_INTERNAL_PROPERTY_PRIMITIVE_BOOLEAN_VALUE, /**< [[Primitive value]] for Boolean objects */
+  ECMA_INTERNAL_PROPERTY_ECMA_VALUE, /**< ecma-value except object */
```
Comment (akiss77) [inline jerry-core/ecma/base/ecma-globals.h]: Was the [[Primitive vaue]] part intentionally removed from the comment? I thought it referred to the standard (8.6.2 Object Internal Properties and Methods, Table 9 - Internal Properties Only Defined for Some Objects).

**Comment by akiss77** [comment]
After the comment fixes, LGTM

**Pull request merged**